File: textFormatting/honeyCombPattern/cVersion/HoneyComb.c
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>

#include "../../../cHeaders/VariableConstants.h"

// gcc HoneyComb.c -o HoneyComb && ./HoneyComb

int main(int argc, char *argv[]) {
    char firstPattern[] = "/ \\_";
    char secondPattern[] = "\\_/ ";

    int toggle = 0;
    int width = 10;
    int height = 20;

    if (argc >= 3) {
        width = atoi(argv[1]);
        height = atoi(argv[2]);
    }

    for (int i = 0; i < height * 2; i++) {
        printf("%c", LINEFEED);
        for (int j = 0; j < width; j++) {
            switch (toggle) {
                case 0 : 
                    printf("%s", firstPattern);
                    break;
                case 1 : 
                    printf("%s", secondPattern);
                    break;
            }
        }
        toggle = 1 - toggle;
    }
}

File: textFormatting/honeyCombPattern/javaVersion/HoneyComb.java
public class HoneyComb {

    private int width = 10;
    private int height  = 10;

    private int toggle = 0;

    private String firstPattern = "/ \\_";
    private String secondPattern = "\\_/ ";

    public HoneyComb(int width, int height) {
        this.width = width;
        this.height = height;
    }
    
    public void run() {
        print();
    }

    public void print() {
        for (int i = 0; i < height * 2; i++) {
            System.out.println("");
            for (int j = 0; j < width; j++) {
                switch (toggle) {
                    case 0 -> System.out.print(firstPattern);
                    case 1 -> System.out.print(secondPattern);
                }
            }
            toggle = 1 - toggle;
        }
    }
}
